jerry-core/jerry.cpp

```diff
@@ -899,6 +899,64 @@ bool jerry_api_get_object_field_value (jerry_api_object_t *object_p,
 }
 
 /**
+ * Applies the given function to the every fields in the objects
+ *
+ */
+void
+jerry_api_foreach_object_field(jerry_api_object_t *object_p, /**< object */
+                               jerry_object_field_foreach_t foreach_p, /**< foreach function */
+                               void *user_data) /**< user data for foreach function */
+{
+  jerry_assert_api_available ();
+
+  for (ecma_property_t *property_p = ecma_get_property_list (object_p);
+       property_p != NULL;
+       property_p = ECMA_GET_POINTER (ecma_property_t, property_p->next_property_p))
+  {
+    if (!ecma_is_property_enumerable (property_p)) continue;
```

Comment:
If someone calls this (the api_foreach) function to a built-in object, then the `ecma_is_property_enumerable` can assert on internal properties. Please move this after the if-elseif-else block below.

Additionally format it like this:
```js
if (!ecma....(property_))
{
  continue;
}
```